go_lib/hooks/tls_certificate/internal_tls.go

```diff
@@ -108,14 +110,24 @@ type GenSelfSignedTLSHookConf struct {
 	BeforeHookCheck func(input *go_hook.HookInput) bool
 }
 
-func (gss GenSelfSignedTLSHookConf) generatePaths() (caPath, certPath, keyPath string) {
-	prefix := strings.TrimSuffix(gss.FullValuesPathPrefix, ".")
+func (gss GenSelfSignedTLSHookConf) path() string {
+	return strings.TrimSuffix(gss.FullValuesPathPrefix, ".")
+}
 
-	caPath = strings.Join([]string{prefix, "ca"}, ".")
-	certPath = strings.Join([]string{prefix, "crt"}, ".")
-	keyPath = strings.Join([]string{prefix, "key"}, ".")
+type certValues struct {
+	Ca  string `json:"ca"`
```

Comment:
Are you sure it isn't `CA`?

Reply:
A little bit more idiomatic, indeed. Changed it.